### PR TITLE
Remove asset pipelin

### LINKED ISSUE #215: Remove asset pipeline support
The asset pipeline support should be removed. See https://help.nextcloud.com/t/asset-pipeline-enabled-leads-to-a-blank-calendar-with-error/1122/5 for details.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #215

The asset pipeline has shown to do more harm than good. Some apps fail
hard with it. Also it makes sure that you download a huge file on each
unvisited page.

@LukasReschke as discussed

CC: @MorrisJobke @BernhardPosselt @popov-d @nickvergessen

### CODE REVIEW

File: lib/private/TemplateLayout.php [hunk 1]
```diff
@@ -137,36 +137,30 @@ public function __construct( $renderAs, $appId = '' ) {
...
 
-		$useAssetPipeline = self::isAssetPipelineEnabled();
-		if ($useAssetPipeline) {
+		// Add the js files
+		$jsFiles = self::findJavascriptFiles(\OC_Util::$scripts);
+		$this->assign('jsfiles', array());
+		if ($this->config->getSystemValue('installed', false) && $renderAs != 'error') {
 			$this->append( 'jsfiles', \OC::$server->getURLGenerator()->linkToRoute('js_config', ['v' => self::$versionHash]));
-			$this->generateAssets();
-		} else {
```
Comment (BernhardPosselt) [inline lib/private/TemplateLayout.php]: Is this else branch only related to assets?

**Comment by BernhardPosselt** [comment]
:+1: looks good, will probably also need a PR to remove the thirdparty libs, correct?

File: lib/private/TemplateLayout.php (hunk 1 above)
Comment (rullzer) [inline lib/private/TemplateLayout.php]: Yes. See the file before the commit. there is the

```php
if ($useAssetPipeline) {
```

**Comment by rullzer** [comment]
Yes will fix 3rdparty stuff after this is in.

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Comment by nickvergessen** [comment]
Btw do we already have a list of important release highlights/changelog for admins?
This should definetly be on it.

**Comment by MorrisJobke** [comment]
cc @jospoortvliet

**Comment by axllent** [comment]
The [online docs](https://docs.nextcloud.com/server/10/admin_manual/configuration_server/js_css_asset_management_configuration.html) need to get updated too as that had me searching for 30 minutes as to why it wasn't working after my upgrade. The option has been removed from the sample config, but the docs don't.

**Comment by jospoortvliet** [comment]
So, we have release notes, I think it'd be nicer to put that in in the future directly rather than wait. It's just a doc in our documentation repo: https://github.com/nextcloud/documentation/edit/stable11/admin_manual/release_notes.rst - easy to edit.

What I don't know is WHAT I should write about this thing in there ;-)

Can someone enlighten or, better, edit the document via the link above and just add a bulletpoint somewhere? No need to make it pretty or perfect, that part I can do...

**Comment by MorrisJobke** [comment]
> So, we have release notes, I think it'd be nicer to put that in in the future directly rather than wait. It's just a doc in our documentation repo: https://github.com/nextcloud/documentation/edit/stable11/admin_manual/release_notes.rst - easy to edit.
> 
> What I don't know is WHAT I should write about this thing in there ;-)
> 
> Can someone enlighten or, better, edit the document via the link above and just add a bulletpoint somewhere? No need to make it pretty or perfect, that part I can do...

cc @rullzer @LukasReschke

**Pull request merged**